test/e2e/azure_test.go

```diff
@@ -468,6 +468,54 @@ var _ = Describe("Workload cluster creation", func() {
 		})
 	})
 
+	// ci-e2e.sh and Prow CI skip this test by default.
+	// To include this test, set `GINKGO_SKIP=""`.
+	Context("Creating a cluster with user-assigned identity [OPTIONAL]", func() {
+		It("with a 1 control plane nodes and 0 worker nodes", func() {
+			clusterName = getClusterName(clusterNamePrefix, "uai")
+
+			// Set CI_RG to a resource group containing a user-assigned identity
+			// named "cloud-provider-user-identity". Defaults to "capz-ci".
+			cirg := os.Getenv("CI_RG")
+			if cirg == "" {
+				cirg = "capz-ci"
+			}
+			Expect(os.Setenv("CI_RG", cirg)).To(Succeed())
+
+			clusterctl.ApplyClusterTemplateAndWait(ctx, clusterctl.ApplyClusterTemplateAndWaitInput{
+				ClusterProxy: bootstrapClusterProxy,
+				ConfigCluster: clusterctl.ConfigClusterInput{
+					LogFolder:                filepath.Join(artifactFolder, "clusters", bootstrapClusterProxy.GetName()),
+					ClusterctlConfigPath:     clusterctlConfigPath,
+					KubeconfigPath:           bootstrapClusterProxy.GetKubeconfigPath(),
+					InfrastructureProvider:   clusterctl.DefaultInfrastructureProvider,
+					Flavor:                   "user-assigned-identity",
+					Namespace:                namespace.Name,
+					ClusterName:              clusterName,
+					KubernetesVersion:        e2eConfig.GetVariable(capi_e2e.KubernetesVersion),
+					ControlPlaneMachineCount: pointer.Int64Ptr(1),
+					WorkerMachineCount:       pointer.Int64Ptr(0),
+				},
+				WaitForClusterIntervals:      e2eConfig.GetIntervals(specName, "wait-cluster"),
+				WaitForControlPlaneIntervals: e2eConfig.GetIntervals(specName, "wait-control-plane"),
+				WaitForMachinePools:          e2eConfig.GetIntervals(specName, "wait-machine-pool-nodes"),
+				ControlPlaneWaiters: clusterctl.ControlPlaneWaiters{
+					WaitForControlPlaneInitialized: InstallCloudProviderAzureHelmChart,
+				},
+			}, result)
+
+			Context("Validating the azure.json secret", func() {
+				AzureJSONSecretSpec(ctx, func() AzureJSONSecretSpecInput {
```

Comment:
instead of validating the contents of the secret, have you considered reusing the LB test spec to expose a service with cloud provider and make sure it is able to use those credentials to create an Azure LB as expected?

Reply:
I had not, but that would be a more durable test that wouldn't have to be changed when we revert the rest. Great idea, I'll give it a try.